### DISCUSSION TITLE
ChannelError.ioOnClosedChannel

### DISCUSSION THREAD

**Discussion by JunyuKuang** [swift-server-community/APNSwift#121] [Q&A]
Hello, 

I'm getting `ChannelError.ioOnClosedChannel` about every 20~30 mins when sending a push.
Is this an expected behavior? 

Currently whenever I receive this error, I create a new `APNSwiftConnection` and close the old one. So far so good, but I'm worrying that Apple may blocks my server, as mentioned in [Apple doc](https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/handling_notification_responses_from_apns?language=objc):

> If your provider server opens and closes its connection to APNs repeatedly, APNs may treat it as a denial-of-service attack and temporarily block your server from connecting.

**Comment by kylebrowning** [comment]
This is not expected, but very old question. Try upgrading to the latest!
